### DISCUSSION TITLE
Best hours for continous consumptions

### DISCUSSION THREAD

**Discussion by fursov** [JaccoR/hass-entso-e#73] [Show and tell]
While doing own home automation I realized that I want to have my water heater and car chargers to run at best times. Also I wanted that those hours are more or less continuous so I do not turn relays on-off all the time. Typically I would need to do these actions ones a day and this nicely falls into a price pattern that we have at least in Finland: prices are cheaper at night for 5-6 hours and then the prices are way higher in day time.

After I have seen this plugin I realized that it should be easy to add few other attributes to the code that would allow to use best time windows for such cases. The code is now in [PR#52](https://github.com/JaccoR/hass-entso-e/pull/52).

There are few sensors and attributes added:
- sensors with names like `time_of_lowest_price_window_for_X_hours_use` where X is in range 2 to 5 and stands for start times of windows with best prices for X hours
- attributes `best_prices` with time span from 1 to 6 hours and 30 minutes step; each attribute entry includes length
of the window `window`, start time of the window `time`, total accumulated price `total`, average price
during the period `average` and average price outside of the period `average_other_time`. The latter can be used to estimate too high prices.

In the PR I have updated readme and also added couple of sample automations that can help to understand the goal of the change (examples folder). I have been using the addition for about two weeks at own home, so not very much own experience.

I was wondering what people think about proposed change and how useful it can be?

**Comment by overas** [comment]
The proposed change is useful for me. 
I have heating cables which I want to control for adjusting target temperature so that it will heat more just before prices will go up (to allow heating less when prices are high), and heat less just before prices will go down. Your example for Room heating does this I think without tried it. The automation for Water heater is also very relevant for me. I hope your work will be included. Thanks for your good work.
